nomad/worker.go

```diff
@@ -215,9 +215,20 @@ func (w *Worker) waitForIndex(index uint64, timeout time.Duration) error {
 	start := time.Now()
 	defer metrics.MeasureSince([]string{"nomad", "worker", "wait_for_index"}, start)
 CHECK:
+	// Snapshot the current state
+	snap, err := w.srv.fsm.State().Snapshot()
```

Comment:
That would be cool! We currently don't have triggers on the index table, so you would have to set up a watch on all the tables. Will put a comment tho